### PR TITLE
Rename SDL frontend

### PR DESCRIPTION
**Pull request by CaptV0rt3x**
* Renamed the SDL frontend from `citra` to `citra-cmd`. (inspired from yuzu)
* Built and tested on Ubuntu 18.04.
* Adds a stub executable named `citra` (based on SDL) which pops up with a message for users. (which needs to be removed from repo after a while)
ytr

- [ ] needs testing on other OS and also CI checks.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3961)

### CODE REVIEW

**Comment by chris062689** [comment]
Please be sure this passes tests related to the installer and start menu. We do not want to break existing users.

File: appveyor.yml [hunk 1]
```diff
@@ -123,7 +123,7 @@ after_build:
           # copy all the dll dependencies to the release folder
           . "./.appveyor/FindDependencies.ps1"
           $DLLSearchPath = "C:\msys64\mingw64\bin;$env:PATH"
-          $MingwDLLs = RecursivelyGetDeps $DLLSearchPath "$RELEASE_DIST\citra.exe"
+          $MingwDLLs = RecursivelyGetDeps $DLLSearchPath "$RELEASE_DIST\citra_cmd.exe"
```
Comment (zhaowenlan1779) [inline appveyor.yml]: This should be "citra-cmd.exe"?

**Comment by zhaowenlan1779** [comment]
I don't think we should break users' installs silently like this, even for the few users that use the SDL frontend. This is kind of bad UX. Some users would link to the SDL frontend for shortcuts to launch games. Perhaps we can leave a symlink of "citra" to "citra-cmd", with additional tip/popup like: "we are renaming the sdl frontend, please change the path of the shortcut" etc. And after some time we delete the symlink and the tip.

**Comment by zhaowenlan1779** [comment]
I don't think we should break users' installs silently like this, even for the few users that use the SDL frontend. This is kind of bad UX. Some users would link to the SDL frontend for shortcuts to launch games. Perhaps we can leave a symlink of "citra" to "citra-cmd", with additional tip/popup like: "we are renaming the sdl frontend, please change the path of the shortcut" etc. And after some time we delete the symlink and the tip.

File: src/citra/CMakeLists.txt [hunk 2]
```diff
@@ -1,30 +1,7 @@
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/CMakeModules)
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR/CMakeModules})
```
Comment (zhaowenlan1779) [inline src/citra/CMakeLists.txt]: Hmm? It should be the original one

File: src/citra/CMakeLists.txt [hunk 3]
```diff
@@ -1,30 +1,7 @@
...
-
-if(UNIX AND NOT APPLE)
-    install(TARGETS citra RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/bin")
-endif()
-
-if (MSVC)
-    include(CopyCitraSDLDeps)
-    copy_citra_SDL_deps(citra)
-endif()
+target_link_libraries(citra PRIVATE Qt5::Widgets)
```
Comment (zhaowenlan1779) [inline src/citra/CMakeLists.txt]: missing newline

File: src/citra/citra.cpp [hunk 4]
```diff
@@ -1,335 +1,31 @@
...
-    }
 
-    while (emu_window->IsOpen()) {
-        system.RunLoop();
-    }
+    QMessageBox::warning(nullptr, "Citra - Application Moved", msg);
 
     return 0;
-}
+}
```
Comment (zhaowenlan1779) [inline src/citra/citra.cpp]: Missing newline

File: src/citra/citra.cpp [hunk 5]
```diff
@@ -1,335 +1,31 @@
...
-// This needs to be included before getopt.h because the latter #defines symbols used by it
-#include "common/microprofile.h"
-
-#include <getopt.h>
-#ifndef _MSC_VER
-#include <unistd.h>
-#endif
+#include <QMessageBox>
+#include <QtGui>
+#include <QtWidgets>
```
Comment (zhaowenlan1779) [inline src/citra/citra.cpp]: I do think you included too much for displaying a message box..
Only `<QApplicaton>` and `<QMessageBox>` are enough according to my test.

File: src/citra/citra.cpp [hunk 6]
```diff
@@ -1,335 +1,28 @@
...
-        } else {
-            LOG_ERROR(Network, "Could not access RoomMember");
-            return 0;
-        }
+    const SDL_MessageBoxData messageboxdata = {
+        SDL_MESSAGEBOX_INFORMATION, /* .flags */
+        NULL,                       /* .window */
+        "Citra - Warning",          /* .title */
+        "\"citra.exe\" has been renamed to \"citra-cmd.exe\". Please update your "
+        "shorcuts!",            /* .message */
```
Comment (zhaowenlan1779) [inline src/citra/citra.cpp]: Hmm? Wasn't you using some condition last time? Why is it "exe" now?

File: src/citra/CMakeLists.txt [hunk 7]
```diff
@@ -1,28 +1,14 @@
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/CMakeModules)
 
+
```
Comment (zhaowenlan1779) [inline src/citra/CMakeLists.txt]: what is this newline for?

File: src/CMakeLists.txt [hunk 8]
```diff
@@ -9,9 +9,10 @@ add_subdirectory(network)
 add_subdirectory(input_common)
 add_subdirectory(tests)
 if (ENABLE_SDL2)
-    add_subdirectory(citra)
+    add_subdirectory(citra_cmd)
 endif()
 if (ENABLE_QT)
+    add_subdirectory(citra)
     add_subdirectory(citra_qt)
```
Comment (zhaowenlan1779) [inline src/CMakeLists.txt]: iirc now you are using sdl2 for citra, why still group it in ENABLE_QT?

File: src/citra/CMakeLists.txt [hunk 9]
```diff
@@ -1,28 +1,14 @@
...
-target_link_libraries(citra PRIVATE common core input_common network)
-target_link_libraries(citra PRIVATE inih glad)
 if (MSVC)
-    target_link_libraries(citra PRIVATE getopt)
-endif()
-target_link_libraries(citra PRIVATE ${PLATFORM_LIBRARIES} SDL2 Threads::Threads)
-
-if(UNIX AND NOT APPLE)
-    install(TARGETS citra RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/bin")
+    target_link_libraries(citra-cmd PRIVATE getopt)
```
Comment (zhaowenlan1779) [inline src/citra/CMakeLists.txt]: Hmm? What for?
Why are you linking libraries to citra-cmd in citra's CMakeLists?

File: src/citra/citra.cpp [hunk 10]
```diff
@@ -1,335 +1,12 @@
-// Copyright 2014 Citra Emulator Project
-// Licensed under GPLv2 or any later version
-// Refer to the license.txt file included.
```
Comment (zhaowenlan1779) [inline src/citra/citra.cpp]: missing copyright notice

File: src/citra/citra.cpp [hunk 11]
```diff
@@ -1,335 +1,12 @@
...
-            LOG_ERROR(Network, "Could not access RoomMember");
-            return 0;
-        }
-    }
-
-    while (emu_window->IsOpen()) {
-        system.RunLoop();
-    }
-
+    SDL_ShowSimpleMessageBox(SDL_MESSAGEBOX_ERROR, "Citra - ERROR", msg, NULL);
```
Comment (zhaowenlan1779) [inline src/citra/citra.cpp]: use nullptr

File: appveyor.yml [hunk 12]
```diff
@@ -123,7 +123,7 @@ after_build:
           # copy all the dll dependencies to the release folder
           . "./.appveyor/FindDependencies.ps1"
           $DLLSearchPath = "C:\msys64\mingw64\bin;$env:PATH"
-          $MingwDLLs = RecursivelyGetDeps $DLLSearchPath "$RELEASE_DIST\citra.exe"
```
Comment (CaptV0rt3x) [inline appveyor.yml]: Should this still be here?

Comment (wwylele) [reply appveyor.yml]: Yes

File: src/citra/CMakeLists.txt [hunk 13]
```diff
@@ -1,28 +1,13 @@
...
-    resource.h
+    stub.cpp
 )
 
-create_target_directory_groups(citra)
-
-target_link_libraries(citra PRIVATE common core input_common network)
-target_link_libraries(citra PRIVATE inih glad)
 if (MSVC)
     target_link_libraries(citra PRIVATE getopt)
```
Comment (wwylele) [inline src/citra/CMakeLists.txt]: no need for this

File: appveyor.yml (hunk 12 above)
Comment (zhaowenlan1779) [reply appveyor.yml]: Hmm? Why? it is just a stub, and all its dependencies should be covered by citra-cmd

**Comment by jroweboy** [comment]
I'm pretty sure we can update the installer to edit the shortcuts and point them to the new locations instead. Of course figuring out how to do that is a pain, so I'm thinking we'll instead just wait for lift installer to get completed and tested out with yuzu, and then just do the whole migration all at once.

Thanks for the contribution!

**Pull request closed without merging**